app/ydoc-shared/src/util/data/result.ts

```diff
@@ -60,10 +60,18 @@ export function unwrap<T, E>(result: Result<T, E>): T {
   else throw result.error
 }
 
+export function unwrapOr<T, A>(result: Result<T, unknown>, alternative: A): T | A
+export function unwrapOr<T, A>(
+  result: Result<T, unknown> | undefined,
+  alternative: A,
+): T | A | undefined
 /** Unwraps the {@link Result} value. If the result is error, an alternative is returned. */
-export function unwrapOr<T, A>(result: Result<T, unknown>, alternative: A): T | A {
-  if (result.ok) return result.value
-  else return alternative
+export function unwrapOr<T, A>(
+  result: Result<T, unknown> | undefined,
+  alternative: A,
+): T | A | undefined {
+  if (!result) return
```

Comment:
This is surprising to me. I'd expect the case with `undefined` result to return `alternative`, similarily as how added `mapOkOr` behaves.